content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -275,33 +275,79 @@ node {
 }
 ----
 
-=== Defining steps
 
+=== Defining global variables
+
+Internally, scripts in the `vars` directory are instantiated as a singleton
+on-demand, when used first.  So it is possible to define more methods,
+properties on a single file that interact with each other:
+
+[source,groovy]
+----
+// vars/acme.groovy
+def setFoo(v) {
+    this.foo = v;
```

Comment:
@daniel-beck http://docs.groovy-lang.org/latest/html/documentation/#properties - see the third example. 
"`this.name` will directly access the field because the property is accessed from within the class that defines it"